src/python/foglamp/core/server.py

```diff
@@ -40,15 +48,77 @@ def _make_app():
         routes.setup(app)
         return app
 
+    @staticmethod
+    def _make_core_app():
+        """Creates the Service management REST server Core a.k.s service registry
+
+        :rtype: web.Application
+        """
+        app = web.Application(middlewares=[middleware.error_middleware])
+        routes.core_setup(app)
+        return app
+
     @classmethod
     async def _start_scheduler(cls):
         """Starts the scheduler"""
         cls.scheduler = Scheduler()
         await cls.scheduler.start()
 
     @classmethod
+    async def start_storage(cls, loop, host, m_port):
+        loop.call_soon(cls._start_storage(host, m_port))
+
+    @staticmethod
+    def _start_storage(host, m_port):
+        print("called _start_storage")
+
+        try:
+            cmd_with_args = ['./storage', '--address={}'.format(host),
+                             '--port={}'.format(m_port)]
+            subprocess.call(cmd_with_args, cwd=_STORAGE_DIR)
+        except Exception as ex:
+            _logger.exception(str(ex))
+
+    @classmethod
+    def _start_core(cls, host, management_port):
+        print("called _start_core")
+        # https://aiohttp.readthedocs.io/en/stable/_modules/aiohttp/web.html#run_app
+        # web.run_app(cls._make_core_app(), host='0.0.0.0', port=8082)
+
+        ma = MultiApp()
+        ma.configure_app(cls._make_core_app(), host=host, port=management_port)
+
+        # TODO: fetch from app info
+        # to register to core; as there is no update mechanism so just register once
+        # with service port
+        service_port = cls.request_available_port()
```

Comment:
This request available port is a little dangerous as the port may be taken in the interim, why can't we simply bind to port 0?